### PR TITLE
Feature/backend/imap outbound

### PR DESCRIPTION
**Pull request by sapiens-sapide**
Users will be able to send emails via their remote identities (IMAP). To do that they must enter credentials for their outbound email server.

- Key for `UserIdentity.Infos["server"]` has been renamed `UserIdentity.Infos["inserver"]` and a new key is expected for sending email : `UserIdentity.Infos["outserver"]`
- Keys in `UserIdentity.Credentials` have been renamed also : 
```go
Credentials{
"password": "xxxx",
"username": "xxxx",
}
```
became  
```go
Credentials{
"inpassword": "xxxx",
"inusername": "xxxx",
"outpassword": "xxxx",
"outusername": "xxxx",
}
```
- A new gocli command has been added to rename keys in current db for inbound server and credentials : `gocli credentialsKeysMigration`
- **NB** : lmtpd needs to access users' credentials to be able to send email via their smtp account. Consequently, a new user "lmtpd" must be added to vault, with `worker` policy rights.

### CODE REVIEW

**Comment by iamdey** [comment]
to merge here https://github.com/CaliOpen/Caliopen/pull/1046 before merging into develop

File: src/backend/configs/apiv2.yaml [hunk 1]
```diff
@@ -18,7 +18,7 @@ ApiConfig:
         buckets:
           raw_messages: caliopen-raw-messages                # bucket name to put raw messages to
           temporary_attachments: caliopen-tmp-attachments    # bucket name to store draft attachments
-      use_vault: false
+      use_vault: true
```
Comment (gdchamal) [inline src/backend/configs/apiv2.yaml]: Can't be set to true for development platform, credentials are in stored in user identity on such platform

File: src/backend/protocols/go.imap/worker.go [hunk 2]
```diff
@@ -83,12 +84,17 @@ func NewWorker(config WorkerConfig, id uint8) (worker *Worker, err error) {
...
 }
 
 func (worker *Worker) Start() error {
 	var err error
-	(*worker).NatsSub, err = worker.NatsConn.QueueSubscribe(worker.Config.NatsTopic, worker.Config.NatsQueue, worker.natsMsgHandler)
+	(*worker).NatsSubs[0], err = worker.NatsConn.QueueSubscribe(worker.Config.NatsTopicFetcher, worker.Config.NatsQueue, worker.natsMsgHandler)
+	if err != nil {
+		return err
+	}
+	(*worker).NatsSubs[1], err = worker.NatsConn.QueueSubscribe(worker.Config.NatsTopicSender, worker.Config.NatsQueue, worker.natsMsgHandler)
```
Comment (gdchamal) [inline src/backend/protocols/go.imap/worker.go]: why subscribe twice on the same queue ?

File: src/backend/configs/apiv2.yaml (hunk 1 above)
Comment (sapiens-sapide) [reply src/backend/configs/apiv2.yaml]: my mistake, it was for testing purpose and I forgot to revert…

File: src/backend/protocols/go.imap/worker.go (hunk 2 above)
Comment (sapiens-sapide) [reply src/backend/protocols/go.imap/worker.go]: that's actualy two different subjects : Fetcher and Sender

Comment (gdchamal) [reply src/backend/protocols/go.imap/worker.go]: Ok my bad

**Pull request merged**